moh_ingest.py

```diff
@@ -0,0 +1,248 @@
+                break
+    else:
+        ingest_finished = True
+
+    if ingest_finished:
+        print("All files have been processed.")
+    else:
+        print("Aborting processing due to an error.")
+
+
+def run_check(katsu_server_url, env_str, data_location, headers, ingest_version):
+    """
+    Run a series of checks to ensure that the ingest is ready to run.
+        - Check if the environment file exists
+        - Check if the environment variable is set
+        - Check if the Katsu server is running the correct version
+        - Check header authentication
+    """
+    # Check if environment file exists
+    if os.path.exists(env_str):
+        print("PASS: The environment file exists.")
+    else:
+        print("ERROR ENV CHECK: The environment file does not exist.")
+
+    # Check if environment variable is set
+    if data_location:
+        print("PASS: Data location is set.")
+    else:
+        print("ERROR LOCATION CHECK: data location is not set.")
+
+    # check if Katsu server is running correct version
+    version_check_url = katsu_server_url + "/api/v1/version_check"
+    try:
+        response = requests.get(version_check_url)
+        if response.status_code == HTTPStatus.OK:
+            katsu_version = response.json()["version"]
+            if check_api_version(
+                ingest_version=ingest_version, katsu_version=katsu_version
+            ):
+                print(f"PASS: Katsu server is running on a compatible version.")
+            else:
+                print(
+                    f"ERROR: Katsu server is running on {katsu_version}. Required version {ingest_version} or greater."
+                )
+        else:
+            print(f"ERROR VERSION CHECK {response.status_code}: {response.text}")
+    except ConnectionError as e:
+        print(f"ERROR VERSION CHECK: {e}")
+        return
+
+    # check authorization
+    if headers == "GET_AUTH_HEADER":
+        try:
+            headers = auth.get_auth_header()
+            print("PASS: Auth header is set.")
+        except Exception as e:
+            print(f"ERROR AUTH CHECK: {e}")
+
+
+def main():
+    print("Select an option:")
+    print("1. Run check")
+    print("2. Ingest data")
+    print("3. Clean data")
+    print("4. Exit")
+
+    # NOTE: FOR DEVELOPMENT ONLY: if you have a local Katsu running and
+    # doesn't want to use auth stack, uncomment the lines below and
+    # comment out the lines after that
+    # katsu_server_url = "http://127.0.0.1:8000"
+    # headers = {"Content-Type": "application/json"}
+    # data_location = "https://raw.githubusercontent.com/CanDIG/katsu/sonchau/moh_part_22/chord_metadata_service/mohpackets/data/small_dataset/synthetic_data/"
+    katsu_server_url = os.environ.get("CANDIG_URL") + "/katsu"
+    headers = "GET_AUTH_HEADER"
+    data_location = os.environ.get("MOH_DATA_LOCATION")
+
+    env_str = "env.sh"
+    ingest_version = "0.9.0"
+
+    choice = int(input("Enter your choice [1-3]: "))
```

Comment:
The use of `input()` here prevents this from being easily automated. This isn't super important, but I recommend bypassing this if you provide command line arguments (using `argparse` to parse them).

Reply:
good point